sdk/resource/process.go

```diff
@@ -39,7 +39,12 @@ var (
 	defaultExecutablePathProvider executablePathProvider = os.Executable
 	defaultCommandArgsProvider    commandArgsProvider    = func() []string { return os.Args }
 	defaultOwnerProvider          ownerProvider          = user.Current
-	defaultRuntimeNameProvider    runtimeNameProvider    = func() string { return runtime.Compiler }
+	defaultRuntimeNameProvider    runtimeNameProvider    = func() string {
+		if runtime.Compiler == "gc" {
+			return "go"
+		}
+		return runtime.Compiler
+	}
```

Comment:
I don't think we should be pulling the runtime name from the Compiler.  There are two compiler frontends, but both of them compile the same source to generate the "go" runtime.  
I think until there is a request from an alternative implementation (the embedded world **might** have a different runtime), it's fine to hardcode this as "go"

Reply:
That works for me.

Reply:
This is what the spec decided we should do.  It's "should", so I guess we're not forced to do it, but I think that if we're not going to do what the spec says we should just continue using `runtime.Compiler` directly as that's the way to get information into that field rather than just filling bits with constant data.

If we're going to have a constant value we should have no value at all since I'm not sure what value it would provide to end users.  The telemetry SDK resource attributes already indicate that the data are coming from the Go SDK.

Reply:
Do we intend not to support `gccgo`? 

FYI: https://github.com/gcc-mirror/gcc/blob/16e2427f50c208dfe07d07f18009969502c25dc8/libgo/go/runtime/compiler.go#L13

Reply:
Technically we don't test against it, so it's not that we don't support it but we don't know if we **do** support gccgo or tinygo.

I find this to be troublesome in a few ways.
1. The spec has code written directly into it.  This may seem like a trivial case, but I think that defiantly breaks the idea of a specification.
2. If we are going to accept that `gc` is not what users (of any of the go's) expect, then why would `gccgo` or `tinygo` be any better.

My argument is that we don't need anything more than `go` there until there is a demand for it.

Reply:
There may be differences in behavior between the same program compiled with `gc`, `gccgo`, and `tinygo`.  If this field does not provide a way to distinguish among them then it seems to me that it is useless and should not be emitted.

Reply:
So I've been thinking about this too, and for what it's worth, I'm leaning towards @Aneurysm9's opinion. Even though not all compilers are being tested at the moment, setting a static value is the same than not setting anything.
If I were to use multiple compilers, I'd be interested in knowing which is which, not having a global `go` value.

Reply:
To make sure this PR doesn't stall, is the plan to revert back to the original approach?

cc @dmathieu @MadVikingGod @Aneurysm9

Reply:
I've gone back to the original approach.

Reply:
Sorry, it took me so long to get back to this.

Yes, I still don't think it is the right choice for this, but I won't stop it.